### PR TITLE
FEATURE: Add support for asset sources in asset editor

### PR DESCRIPTION
**Pull request by robertlemke**
This change adds asset sources support to the Asset Editor, which
allows for searching and importing remote assets on the fly.

This change relies on the new Asset Proxies web service endpoint in Neos.Neos.

https://github.com/neos/neos-development-collection/issues/2198

### CODE REVIEW

**Comment by robertlemke** [comment]
Not finished yet – I'd like to get rid of the duplicate import issue first.

File: packages/neos-ui-editors/src/Editors/AssetEditor/index.js [hunk 1]
```diff
@@ -140,6 +160,7 @@ export default class AssetEditor extends PureComponent {
     }
 
     handleMediaSelected = assetIdentifier => {
+        console.log('handleMediaSelected', assetIdentifier, this.props);
```
Comment (daniellienert) [inline packages/neos-ui-editors/src/Editors/AssetEditor/index.js]: I guess this console log needs to be removed

Comment (robertlemke) [reply packages/neos-ui-editors/src/Editors/AssetEditor/index.js]: Yes indeed.

**Comment by robertlemke** [comment]
I added a filter which removes those assets from a search result which are already selected in the asset editor.

Note that the asset source proxy implementation must be properly implemented so that _$proxy->getLocalAssetIdentifier()_ returns the Neos Media Asset identifier if the asset was already imported (via the _ImportedAssetRepository_).

**Comment by dimaip** [comment]
Hey @robertlemke! What's the current state of this thing? Is it finished and ready for final review?

**Comment by robertlemke** [comment]
Allright, I added some grouping and display of asset source labels, by suggestion of @bwaidelich (see Neos.Neos PR). I suggest that we get the Neos PR merged first, and then this here can be merged.

However, from my side, this should be fine already for a final review.

**Comment by bwaidelich** [comment]
@robertlemke 
> I added some grouping and display of asset source labels
but did you _push_ it as well? :)

**Comment by robertlemke** [comment]
@bwaidelich yes, I did: https://github.com/neos/neos-ui/pull/2161/commits/4033bcab511c4ba7ea665781ce395a7ea3c860fa

**Comment by bwaidelich** [comment]
Ah apparently, Github is still not back to full speed. The commit just appeared now

**Comment by kitsunet** [comment]
@robertlemke @bwaidelich @dimaip This change should be reverted immediately as it (unannouncedly) breaks compatiblity with Neos 3.3 for the UI. Neos 3.3 doesn't support asset proxies so the asset search is done on an empty url (so basically domain/?parameters) so it stays hidden and doesn't produce an obvious error. Asset usage is broken with this though.

**Comment by dimaip** [comment]
Also see our quick discussion https://neos-project.slack.com/archives/C0U0KEGDQ/p1543414898215700

**Comment by robertlemke** [comment]
I think that just reverting the feature in not the solution.

As @dimaip already wrote in Slack, the long-term solution is to make Neos UI a part of the Neos Development Collection. However, since we are not there yet, the logical solution in my opinion is, that we branch off the Neos UI repository in order to provide a backwards-compatible version for older Neos versions. That branch should only receive important bugfixes, all new features should go into master and be compatible with the latest / upcoming Neos.

I don't have insight into how Neos UI is released currently, but I guess it's rather simple, given that it is not part of the Development Collection yet? If so, let's create a branch, revert this feature in that branch and that should be it.

**Comment by robertlemke** [comment]
I think that just reverting the feature in not the solution.

As @dimaip already wrote in Slack, the long-term solution is to make Neos UI a part of the Neos Development Collection. However, since we are not there yet, the logical solution in my opinion is, that we branch off the Neos UI repository in order to provide a backwards-compatible version for older Neos versions. That branch should only receive important bugfixes, all new features should go into master and be compatible with the latest / upcoming Neos.

I don't have insight into how Neos UI is released currently, but I guess it's rather simple, given that it is not part of the Development Collection yet? If so, let's create a branch, revert this feature in that branch and that should be it.

**Comment by kitsunet** [comment]
Totally fine for me, but the branch needs to be maintained as long as 3.3 is maintained IMHO.

**Comment by kitsunet** [comment]
Fact is, we cannot release a new version of the UI right now as that would be a breaking release and I now had to fork the UI for a 3.3 project to make sure we don't run into this.

**Comment by dimaip** [comment]
True. It's one of the solutions. Will try to weigh-in all the pros and cons tomorrow at 13:00 in the hangout with Christian and all those who would want to join.

**Pull request merged**